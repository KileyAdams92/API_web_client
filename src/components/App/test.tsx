import React from "react"
import ReactDOM from "react-dom"
import { Base as App, mapStateToProps } from "."
import { render, fireEvent, cleanup } from "react-testing-library"

afterEach(cleanup)

describe("App", () => {
  it("renders", () => {
    const div = document.createElement("div")
    ReactDOM.render(<App todos={[]} addTodo={jest.fn()} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it("renders expected structure", () => {
    const { container } = render(<App todos={[]} addTodo={jest.fn()} />)
    expect(container).toMatchSnapshot()
  })

  it("renders todos", () => {
    const { container } = render(
      <App
        todos={[
          { id: "1", title: "Test Todo 1", complete: false },
          { id: "2", title: "Test Todo 2", complete: true },
        ]}
        addTodo={jest.fn()}
      />,
    )
    expect(container).toMatchSnapshot()
  })

  it("handles clicking of button", () => {
    // Arrange
    const onClick = jest.fn()
    const { getByText } = render(<App todos={[]} addTodo={onClick} />)

    // Act
    fireEvent.click(getByText(/Add Todo/i))

    // Assert
    expect(onClick).toHaveBeenCalled()
  })
})

describe("App mapStateToProps", () => {
  it("maps state.todos to todos prop", () => {
    expect(
      mapStateToProps({
        todos: {
          entities: {
            "1": { id: "1", title: "Test Todo", complete: false },
          },
          result: ["1"],
        },
      }),
    ).toMatchSnapshot()
  })
})
