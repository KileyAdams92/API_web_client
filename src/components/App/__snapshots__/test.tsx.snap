// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App mapStateToProps maps state.todos to todos prop 1`] = `
Object {
  "todos": Array [
    Object {
      "complete": false,
      "id": "1",
      "title": "Test Todo",
    },
  ],
}
`;

exports[`App renders expected structure 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <p>
        Edit 
        <code>
          src/App.tsx
        </code>
         and save to reload.
      </p>
      <a
        class="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
      <p>
        Todos: 
        0
      </p>
      <button>
        Add Todo
      </button>
      <ul />
    </header>
  </div>
</div>
`;

exports[`App renders todos 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <p>
        Edit 
        <code>
          src/App.tsx
        </code>
         and save to reload.
      </p>
      <a
        class="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
      <p>
        Todos: 
        2
      </p>
      <button>
        Add Todo
      </button>
      <ul>
        <li>
          Test Todo 1
           | 
          Incomplete
        </li>
        <li>
          Test Todo 2
           | 
          Complete
        </li>
      </ul>
    </header>
  </div>
</div>
`;
